src/Flight/FlightService.java

```diff
@@ -1,6 +1,143 @@
+        } else if(!validateMinFlight(flight)){
+            System.out.println("Invalid information provided.");
+        } else {
+            flights[indexToReplace] = flight;
+            System.out.printf("Flight %s successfully added\n", flight);
+        }
+    }
+
+    public Flight findById(int flightNumber){
+        for (Flight flight : flights) {
+            if (flight != null && flight.flightNumber == flightNumber) {
+                return flight;
+            }
+        }
+
+        return null;
+    }
+
+
+    public boolean updateFlight(Flight flightToUpdate, String timeArrival, String timeDeparture, int pilot, int airline) {
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss");
+
+        flightToUpdate.setTimeArrival(LocalDateTime.parse(timeArrival, formatter));
+        flightToUpdate.setTimeDeparture(LocalDateTime.parse(timeDeparture, formatter));
+        flightToUpdate.setPilot(pilot);
+        flightToUpdate.setAirline(airline);
+
+        if(!validateFullFlight(flightToUpdate)) return false;
+
+        return true;
+    }
+
+
+    private boolean isEmpty(Flight[] arr) { // defining the parameter of a string array to be included when executing this mehtod
+        for (Flight element : arr) { // enhanced for each loop, that iterates through the arrays elements returning & assigning the value to the declared variable flight
+            if (element != null) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    public int getFirstNull(Flight[] arr) { //get the first null index to replace, otherwise return -1
+        for (int i = 0; i < arr.length; i++) {
+            if (arr[i] == null) {
+                return i;
+            }
+        }
+        return -1;
+    }
+
+    private boolean validateMinFlight(Flight flight) {
+        if (flight == null) {
+            return false;
+        }
+
+        if (flight.flightNumber < 1000 || flight.flightNumber > 999999) {
+            System.out.println("Invalid flight number");
+            return false;
+        }
+
+        if (flight.originAirport.length() != 3 || flight.destinationAirport.length() != 3) {
+            System.out.println("Invalid airport code");
+            return false;
+        }
+
+        if (flight.seatCount < 0) {
+            System.out.println("Invalid seat count");
+            return false;
+        }
+
+        return true;
+    }
+
+    private static boolean validateFullFlight(Flight flight) {
+        if (flight == null) {
+            return false;
+        }
+        if (flight.flightNumber < 1000 || flight.flightNumber > 999999) {
+            System.out.println("Invalid flight number");
+            return false;
+        }
+
+        if (flight.originAirport.length() != 3 || flight.destinationAirport.length() != 3) {
+            System.out.println("Invalid airport code");
+            return false;
+        }
+
+        if (flight.seatCount < 0) {
```

Comment:
Both here and in the validateMinFlight(Flight flight) methods, it does not check if seatCount is exactly 0, and there is no upper limit on seatCount either. I'm not sure if this seatCount is actually the "maximum seats" or just the "seats remaining". I am curious, is not having an upper limit deliberate or just not seen as a big issue?